string cmd
string cmd_name
string cmd_description
string stdout
string stderr
# ret_val == -1 if it hasn't finished yet
int32 ret_val
time time_running